File: dev/20190822/k2works/src/Application/Command/FizzBuzzListCommand.ts
import { FizzBuzzList } from "../../Domain/Model/FizzBuzzList";
import { IFizzBuzz } from "../Service/IFizzBuzz";
// tslint:disable-next-line: ordered-imports
import { IFizzBuzzCommand } from "../Service/IFizzBuzzCommand";
import { IFizzBuzzType } from "../Service/IFizzBuzzType";

export class FizzBuzzListCommand implements IFizzBuzzCommand {
  // tslint:disable-next-line: variable-name
  private _type: IFizzBuzzType;
  // tslint:disable-next-line: variable-name
  private _list: FizzBuzzList;

  constructor(type: IFizzBuzzType) {
    this._type = type;
  }

  public execute(input: number): IFizzBuzz {
    if (input < 0) {
      throw new Error("FizzBuzzList can't generate by minus number");
    }
    this._list = new FizzBuzzList([]);

    IFizzBuzz.range(1, input).forEach(
      i => (this._list = this._list.add(this._type.generate(i)))
    );

    return this._list;
  }
}

File: dev/20190902/k2works/app/FizzBuzz/Application/Command/FizzBuzzListCommand.cs
using System;
using System.Linq;
using FizzBuzz.Application.Service;
using FizzBuzz.Domain.Model;

namespace FizzBuzz.Application.Command
{
    public class FizzBuzzListCommand: IFizzBuzzCommand
    {
        private IFizzBuzzType _type;

        public FizzBuzzListCommand(IFizzBuzzType type)
        {
            _type = type;
        }

        public IFizzBuzz Execute(int number)
        {
            if (number < 0) throw new ArgumentException("FizzBuzzList can't generate by minus number'");
            return new FizzBuzzList(Enumerable.Range(1, number).Select(i => _type.Generate(i)).ToList());
        }
    }
}